### PR TITLE
Use reusable workflows

### PR DESCRIPTION
**Pull request by dinakar29**
Use centralized Github Action workflows.

### CODE REVIEW

**Review by danielinteractive — APPROVED** [review]
thx @dinakar29 ! looks good. there is some failing latex problem that we did not have before though - maybe something that can be fixed?

**Comment by dinakar29** [comment]
> thx @dinakar29 ! looks good. there is some failing latex problem that we did not have before though - maybe something that can be fixed?

Yes, fixing this now.

**Comment by danielinteractive** [comment]
@dinakar29 not sure what is happening here, but it is strange that only with the workflow update the latex fails right? maybe you can check what is the difference re: latex handling going from individual to reusable workflow? I would prefer not to mess with the contents themselves (as it worked before)

**Comment by dinakar29** [comment]
> @dinakar29 not sure what is happening here, but it is strange that only with the workflow update the latex fails right? maybe you can check what is the difference re: latex handling going from individual to reusable workflow? I would prefer not to mess with the contents themselves (as it worked before)

Yeah I'm looking into the differences right now. The JSSstyle LaTeX package might be out of date with the newer version of TeX, which could explain the underlying issue. I am able to consistently reproduce the issue. I'll provide more details soon as I determine the root cause. 

As a side note - are you intending to keep the old JSS submissions in this repo? There are tarballs, PDFs and LaTeX compile logs and artifacts, which, frankly should not be stored in version control.

**Comment by danielinteractive** [comment]
@dinakar29 you are right - thanks for reminding me of the old baggage. I now pushed a commit here where I deleted a  lot of the old JSS folders and files. maybe that helps?

**Comment by dinakar29** [comment]
> @dinakar29 you are right - thanks for reminding me of the old baggage. I now pushed a commit here where I deleted a lot of the old JSS folders and files. maybe that helps?

Thanks! I'll sort this out. I'm working on identifying and fixing the root cause and will resolve it soon.

File: crmPack.Rproj [hunk 1]
```diff
@@ -9,7 +9,7 @@ UseSpacesForTab: Yes
 NumSpacesForTab: 2
 Encoding: UTF-8
 
-RnwWeave: Sweave
+RnwWeave: knitr
```
Comment (dinakar29) [inline crmPack.Rproj:12]: This makes it convenient to compile the PDF within RStudio

File: vignettes/crmPack-jss-paper.Rnw [hunk 2]
```diff
@@ -27,6 +24,8 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %% declarations for jss.cls %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\definecolor{Red}{rgb}{1.0, 0.0, 0.0}
+\definecolor{Blue}{rgb}{0.0, 0.0, 1.0}
```
Comment (dinakar29) [inline vignettes/crmPack-jss-paper.Rnw:28]: This should resolve the color definitions issue.

**Pull request merged**